nncf/common/pruning/export_helpers.py

```diff
@@ -50,3 +61,212 @@ def get_all_op_aliases(cls):
+
+
+class OpInput(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return False
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        node.data['output_mask'] = None
+
+
+class OpOutput(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        node.data['output_mask'] = None
+
+
+class OpIdentityMaskForwardOps(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        identity_mask_propagation(node, graph)
+
+
+class OpConvolution(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        accept_pruned_input = True
+        if is_grouped_conv(node):
+            if not is_depthwise_conv(node):
+                accept_pruned_input = False
+        return accept_pruned_input
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        input_masks = get_input_masks(node, graph)
+        output_mask = node.data.get('output_mask', None)
+
+        if is_grouped_conv(node):
+            output_mask = None
+            if is_depthwise_conv(node):
+                output_mask = input_masks[0]
+
+        node.data['output_mask'] = output_mask
+
+
+class OpTransposeConvolution(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        accept_pruned_input = True
+        if is_grouped_conv(node):
+            if not is_depthwise_conv(node):
+                accept_pruned_input = False
+        return accept_pruned_input
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        input_masks = get_input_masks(node, graph)
+        output_mask = node.data.get('output_mask', None)
+
+        # In case of group convs we can't prune by output filters
+        if is_grouped_conv(node):
+            output_mask = None
+            if is_depthwise_conv(node):
+                output_mask = input_masks[0]
+
+        node.data['output_mask'] = output_mask
+
+
+class OpBatchNorm(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        identity_mask_propagation(node, graph)
+
+
+class OpGroupNorm(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        # For Instance Normalization
+        return isinstance(node.layer_attributes, GroupNormLayerAttributes) \
+               and node.layer_attributes.num_groups == node.layer_attributes.num_channels
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        identity_mask_propagation(node, graph)
```

Comment:
@mkaglins, could you please tell me, is it valid point that even if `accept_pruned_intput==False` mask porpagated as is?